<a href="https://colab.research.google.com/github/vir097/Cardiovascular-Risk-Prediction--Viral-Shewale/blob/main/Cardiovascular_risk_prediction_Viral_Shewale.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Project Tile: Cardiovascular Risk Prediction**

**Problem Statment:**
The dataset is from an ongoing cardiovascular study on residents of the town of Framingham,
Massachusetts. The classification goal is to predict whether the patient has a 10-year risk of
future coronary heart disease (CHD).

**Data Description**

**Demographic:**
* Sex: male or female("M" or "F")
* Age: Age of the patient;(Continuous - Although the recorded ages have been truncated to
whole numbers, the concept of age is continuous)
Behavioral
* is_smoking: whether or not the patient is a current smoker ("YES" or "NO")
* Cigs Per Day: the number of cigarettes that the person smoked on average in one day.(can be
considered continuous as one can have any number of cigarettes, even half a cigarette.)
Medical( history)
* BP Meds: whether or not the patient was on blood pressure medication (Nominal)
* Prevalent Stroke: whether or not the patient had previously had a stroke (Nominal)
* Prevalent Hyp: whether or not the patient was hypertensive (Nominal)
* Diabetes: whether or not the patient had diabetes (Nominal)
Medical(current)
* Tot Chol: total cholesterol level (Continuous)
* Sys BP: systolic blood pressure (Continuous)
* Dia BP: diastolic blood pressure (Continuous)
* BMI: Body Mass Index (Continuous)
* Heart Rate: heart rate (Continuous - In medical research, variables such as heart rate though in
fact discrete, yet are considered continuous because of large number of possible values.)
* Glucose: glucose level (Continuous)
Predict variable (desired target)
* 10-year risk of coronary heart disease CHD(binary: “1”, means “Yes”, “0” means “No”) -
DV

## **1. Load Dataset**

In [1]:
# import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

%matplotlib inline
sns.set()

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/Viral/Cardiovascular Risk Prediction--Viral Shewale/data_cardiovascular_risk.csv",index_col='id')

## **2. Basic Inspection of the dataset**

In [4]:
# Get the head of the dataset
df.head()

age  education sex is_smoking  cigsPerDay  BPMeds  prevalentStroke  \
id                                                                       
0    64        2.0   F        YES         3.0     0.0                0   
1    36        4.0   M         NO         0.0     0.0                0   
2    46        1.0   F        YES        10.0     0.0                0   
3    50        1.0   M        YES        20.0     0.0                0   
4    64        1.0   F        YES        30.0     0.0                0   

    prevalentHyp  diabetes  totChol  sysBP  diaBP    BMI  heartRate  glucose  \
id                                                                             
0              0         0    221.0  148.0   85.0    NaN       90.0     80.0   
1              1         0    212.0  168.0   98.0  29.77       72.0     75.0   
2              0         0    250.0  116.0   71.0  20.35       88.0     94.0   
3              1         0    233.0  158.0   88.0  28.26       68.0     94.0   
4              0         0    241.0  136.5   85.0  26.42       70.0     77.0   

    TenYearCHD  
id              
0            1  
1            0  
2            0  
3            1  
4            0

In [5]:
# Get the tail of the dataset
df.tail()

age  education sex is_smoking  cigsPerDay  BPMeds  prevalentStroke  \
id                                                                         
3385   60        1.0   F         NO         0.0     0.0                0   
3386   46        1.0   F         NO         0.0     0.0                0   
3387   44        3.0   M        YES         3.0     0.0                0   
3388   60        1.0   M         NO         0.0     NaN                0   
3389   54        3.0   F         NO         0.0     0.0                0   

      prevalentHyp  diabetes  totChol  sysBP  diaBP    BMI  heartRate  \
id                                                                      
3385             0         0    261.0  123.5   79.0  29.28       70.0   
3386             0         0    199.0  102.0   56.0  21.96       80.0   
3387             1         0    352.0  164.0  119.0  28.92       73.0   
3388             1         0    191.0  167.0  105.0  23.01       80.0   
3389             0         0    288.0  124.0   77.0  29.88       79.0   

      glucose  TenYearCHD  
id                         
3385    103.0           0  
3386     84.0           0  
3387     72.0           1  
3388     85.0           0  
3389     92.0           0

In [6]:
# get the shape of the dataset
df.shape

(3390, 16)

* There are $3390$ rows and $16$ different attributes (columns) in the dataset.

In [7]:
# columns in the dataset
df.columns

Index(['age', 'education', 'sex', 'is_smoking', 'cigsPerDay', 'BPMeds',
       'prevalentStroke', 'prevalentHyp', 'diabetes', 'totChol', 'sysBP',
       'diaBP', 'BMI', 'heartRate', 'glucose', 'TenYearCHD'],
      dtype='object')

In [8]:
# Describtion of the dataset
df.describe(include='all')

age    education   sex is_smoking   cigsPerDay       BPMeds  \
count   3390.000000  3303.000000  3390       3390  3368.000000  3346.000000   
unique          NaN          NaN     2          2          NaN          NaN   
top             NaN          NaN     F         NO          NaN          NaN   
freq            NaN          NaN  1923       1703          NaN          NaN   
mean      49.542183     1.970936   NaN        NaN     9.069477     0.029886   
std        8.592878     1.019081   NaN        NaN    11.879078     0.170299   
min       32.000000     1.000000   NaN        NaN     0.000000     0.000000   
25%       42.000000     1.000000   NaN        NaN     0.000000     0.000000   
50%       49.000000     2.000000   NaN        NaN     0.000000     0.000000   
75%       56.000000     3.000000   NaN        NaN    20.000000     0.000000   
max       70.000000     4.000000   NaN        NaN    70.000000     1.000000   

        prevalentStroke  prevalentHyp     diabetes      totChol       sysBP  \
count       3390.000000   3390.000000  3390.000000  3352.000000  3390.00000   
unique              NaN           NaN          NaN          NaN         NaN   
top                 NaN           NaN          NaN          NaN         NaN   
freq                NaN           NaN          NaN          NaN         NaN   
mean           0.006490      0.315339     0.025664   237.074284   132.60118   
std            0.080309      0.464719     0.158153    45.247430    22.29203   
min            0.000000      0.000000     0.000000   107.000000    83.50000   
25%            0.000000      0.000000     0.000000   206.000000   117.00000   
50%            0.000000      0.000000     0.000000   234.000000   128.50000   
75%            0.000000      1.000000     0.000000   264.000000   144.00000   
max            1.000000      1.000000     1.000000   696.000000   295.00000   

              diaBP          BMI    heartRate      glucose   TenYearCHD  
count   3390.000000  3376.000000  3389.000000  3086.000000  3390.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean      82.883038    25.794964    75.977279    82.086520     0.150737  
std       12.023581     4.115449    11.971868    24.244753     0.357846  
min       48.000000    15.960000    45.000000    40.000000     0.000000  
25%       74.500000    23.020000    68.000000    71.000000     0.000000  
50%       82.000000    25.380000    75.000000    78.000000     0.000000  
75%       90.000000    28.040000    83.000000    87.000000     0.000000  
max      142.500000    56.800000   143.000000   394.000000     1.000000

In [9]:
# data type
df.dtypes

age                  int64
education          float64
sex                 object
is_smoking          object
cigsPerDay         float64
BPMeds             float64
prevalentStroke      int64
prevalentHyp         int64
diabetes             int64
totChol            float64
sysBP              float64
diaBP              float64
BMI                float64
heartRate          float64
glucose            float64
TenYearCHD           int64
dtype: object